numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1749,11 +1782,10 @@ unpack_bits(PyObject *input, int axis)
             v |= (npy_uint64)((j &  32) ==  32) << 40;
             v |= (npy_uint64)((j &  64) ==  64) << 48;
             v |= (npy_uint64)((j & 128) == 128) << 56;
-#if NPY_BYTE_ORDER == NPY_LITTLE_ENDIAN
             v = npy_bswap8(v);
-#endif
```

Comment:
I think this was a bug, if not we need to add it back in. I don't have a big-endian system to test on.

Reply:
Hmm, this does get turned into bytes in the end, and those will have a different order in the `uint64` here. So, my guess is that you do need to swap (i.e., have the reverse swap for big and little endian). But am a bit too lazy to just write it out...  Note that the test cases surely would fail for big endian if this was wrong...

Reply:
Now tested on a big-endian system since I have access to the gcc build farm and a sparc64 machine. See 5d1b9208d